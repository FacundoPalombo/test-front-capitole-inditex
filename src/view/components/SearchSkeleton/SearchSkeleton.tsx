import React from 'react';
import Channel from '@containers/search/components/Channel/Channel';
import image from './asset.jpg';
import styles from './styles.module.scss';
import noop from '@utils/noop';

const SearchSkeleton = () => {
  const skeletonMock = 'abcdefghijklmnopqrstu'.split('').map((i) => ({
    image,
    artist: '....................',
    title: '..........................',
    podcastId: i,
    key: i,
    skeleton: true,
  }));

  return (
    <section id="search-skeleton" className={styles.search_skeleton__container}>
      <span className={styles.search_skeleton__search_box}>
        <label
          htmlFor="search_skeleton__search_box"
          aria-label="Filter over all the 100 best podcasts on itunes."
          className={styles.search_skeleton__search_box__label}
        >
          0
        </label>
        <input
          placeholder="..."
          id="search_skeleton__search_box"
          data-testid="search_skeleton__search_box"
          name="search_skeleton__search_box"
          type="text"
          className={styles.search_skeleton__search_box__input}
          onChange={noop}
          value=""
        />
      </span>
      {skeletonMock.map((channel) => (
        <Channel {...channel} />
      ))}
    </section>
  );
};

export default SearchSkeleton;
