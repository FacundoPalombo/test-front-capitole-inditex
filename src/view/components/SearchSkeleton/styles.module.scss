.search_skeleton {
  &__container {
    * {
      /* the important is only for build a skeleton dinamically without a custom implementation of each component to build a individual skeleton*/
      pointer-events: none !important;
      color: transparent;
    }
    h3,
    p {
      background: linear-gradient(
        90deg,
        #fff 0%,
        rgba(113, 113, 113, 0.1),
        #fff 100%
      );
      height: 1.5em;
      background-size: 100px;
      background-repeat: repeat;
      margin: 4px;
      border-radius: 6px;
      animation: 1.2s cubic-bezier(0.03, -0.14, 0.52, 1.3) infinite backwards
        slide;
      @keyframes slide {
        to {
          background-position-x: 250px;
        }
      }
    }
    position: relative;
    padding-top: 140px;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    gap: 16px;
    align-items: center;
    justify-content: center;
  }
  &__search_box {
    position: absolute;
    top: 0;
    right: 0;

    &__input {
      color: var(--font-disabled);
      padding: 8px 8px 8px 12px;
      margin: 16px 8px 8px 16px;
      border: 1px solid var(--font-disabled);
      border-radius: 4px;
      & ::placeholder {
        font-size: small;
        color: var(--font-disabled);
      }
    }

    &__label {
      background-color: var(--font-disabled);
      padding: 2px 8px;
      font-size: large;
      font-weight: 700;
      color: white;
      border-radius: 5px;
    }
  }
}
