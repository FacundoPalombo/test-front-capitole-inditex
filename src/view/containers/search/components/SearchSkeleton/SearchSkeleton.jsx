import React from 'react';
import Song from '../Song';
import SearchBox from '../SearchBox';
import image from './asset.jpg';
import styles from './styles.module.scss';

/* 
# Just ignore this skeleton components

It was only for testing purposes that i made for, because i wanna try some new methods in react router with the suspense features but
for the differences between promises and the boundaries of react, the user experience of loading some components apart and others skeletons apart at the same time looks quite a bit awk.
 */

const SearchSkeleton = () => {
  const skeletonMock = 'abcdefghijklmnopqrstuvwxyz'.split('').map((i) => ({
    image,
    artist: '....',
    title: '....',
    podcastId: i,
    key: i,
    skeleton: true,
  }));
  return (
    <section id="search-skeleton" className={styles.search_skeleton__container}>
      <SearchBox onChange={() => {}} value="" skeleton />
      {skeletonMock.map((song) => (
        <Song {...song} />
      ))}
    </section>
  );
};

export default SearchSkeleton;
