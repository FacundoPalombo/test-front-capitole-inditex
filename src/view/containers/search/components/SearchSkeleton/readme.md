# Just ignore this skeleton components

It was only for testing purposes that i made for, because i wanna try some new methods in react router with the suspense features but
for the differences between promises and the boundaries of react, the user experience of loading some components apart and others skeletons apart at the same time looks quite a bit awk.
