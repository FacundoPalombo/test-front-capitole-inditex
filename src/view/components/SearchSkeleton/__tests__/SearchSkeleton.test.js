import React from 'react';
import { render } from '@testing-library/react';
import SearchSkeleton from '../SearchSkeleton.tsx';

jest.mock('@containers/search/components/Channel/Channel.jsx', () => () => (
  <div id="channel-mock">Hola, soy un item de channel</div>
));

describe('SearchSkeleton unit tests', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<SearchSkeleton />);
    expect(asFragment()).toMatchSnapshot();
  });
});
