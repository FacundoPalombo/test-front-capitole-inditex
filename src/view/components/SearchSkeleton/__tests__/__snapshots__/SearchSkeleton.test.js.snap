// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSkeleton unit tests should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="search_skeleton__container"
    id="search-skeleton"
  >
    <span
      class="search_skeleton__search_box"
    >
      <label
        aria-label="Filter over all the 100 best podcasts on itunes."
        class="search_skeleton__search_box__label"
        for="search_skeleton__search_box"
      >
        0
      </label>
      <input
        class="search_skeleton__search_box__input"
        data-testid="search_skeleton__search_box"
        id="search_skeleton__search_box"
        name="search_skeleton__search_box"
        placeholder="..."
        type="text"
        value=""
      />
    </span>
    <div
      id="channel-mock"
    >
      Hola, soy un item de channel
    </div>
    <div
      id="channel-mock"
    >
      Hola, soy un item de channel
    </div>
    <div
      id="channel-mock"
    >
      Hola, soy un item de channel
    </div>
    <div
      id="channel-mock"
    >
      Hola, soy un item de channel
    </div>
    <div
      id="channel-mock"
    >
      Hola, soy un item de channel
    </div>
    <div
      id="channel-mock"
    >
      Hola, soy un item de channel
    </div>
    <div
      id="channel-mock"
    >
      Hola, soy un item de channel
    </div>
    <div
      id="channel-mock"
    >
      Hola, soy un item de channel
    </div>
    <div
      id="channel-mock"
    >
      Hola, soy un item de channel
    </div>
    <div
      id="channel-mock"
    >
      Hola, soy un item de channel
    </div>
    <div
      id="channel-mock"
    >
      Hola, soy un item de channel
    </div>
    <div
      id="channel-mock"
    >
      Hola, soy un item de channel
    </div>
    <div
      id="channel-mock"
    >
      Hola, soy un item de channel
    </div>
    <div
      id="channel-mock"
    >
      Hola, soy un item de channel
    </div>
    <div
      id="channel-mock"
    >
      Hola, soy un item de channel
    </div>
    <div
      id="channel-mock"
    >
      Hola, soy un item de channel
    </div>
    <div
      id="channel-mock"
    >
      Hola, soy un item de channel
    </div>
    <div
      id="channel-mock"
    >
      Hola, soy un item de channel
    </div>
    <div
      id="channel-mock"
    >
      Hola, soy un item de channel
    </div>
    <div
      id="channel-mock"
    >
      Hola, soy un item de channel
    </div>
    <div
      id="channel-mock"
    >
      Hola, soy un item de channel
    </div>
  </section>
</DocumentFragment>
`;
